Modix.Services/Utilities/RequireModixPermissionAttribute.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Modix.Data.Utilities;
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Modix.Data;
+
+namespace Modix.Services.Utilities
+{
+    public class RequireModixPermissionAttribute : PreconditionAttribute
+    {
+        private readonly Permissions _requiredPermission;
+
+        public RequireModixPermissionAttribute(Permissions requiredPermission)
+        {
+            _requiredPermission = requiredPermission;
+        }
+
+        public override Task<PreconditionResult> CheckPermissions(ICommandContext context, CommandInfo command, IServiceProvider map)
+        {
+            // first check if we're in a guild because this won't work if we aren't in a guild
+            if (!(context.User is IGuildUser))
+                return Task.FromResult(PreconditionResult.FromError("The current context isn't a guild"));
+
+            using (var db = map.GetService<ModixContext>())
```

Comment:
probably should cache, with some way to invalidate the cache when the roles/permissions change.